src/main/java/uk/gov/companieshouse/web/pps/controller/pps/ConfirmationController.java

```diff
@@ -63,28 +80,46 @@ public String getConfirmation(@PathVariable String companyNumber,
             return ERROR_VIEW;
         }
 
-
         // If the payment is anything but paid return user to beginning of journey
+        PayableLateFilingPenalty payableLateFilingPenalty;
+        CompanyProfileApi companyProfileApi;
+        try {
+            companyProfileApi = companyService.getCompanyProfile(companyNumber);
+            payableLateFilingPenalty = payablePenaltyService
+                    .getPayableLateFilingPenalty(companyNumber, penaltyId);
+        } catch (ServiceException ex) {
+            LOGGER.errorRequest(request, ex.getMessage(), ex);
+            return ERROR_VIEW;
+        }
+
         if (!paymentStatus.equals("paid")) {
             LOGGER.info("Payment status is " + paymentStatus + " and not of status 'paid', returning to beginning of journey");
-            PayableLateFilingPenalty payableLateFilingPenalty;
-            try {
-                payableLateFilingPenalty = payablePenaltyService
-                        .getPayableLateFilingPenalty(companyNumber, penaltyId);
-            } catch (ServiceException ex) {
-                LOGGER.errorRequest(request, ex.getMessage(), ex);
-                return ERROR_VIEW;
-            }
-
             Map<String, String> links = payableLateFilingPenalty.getLinks();
             return UrlBasedViewResolver.REDIRECT_URL_PREFIX + links.get("resume_journey_uri");
 
         }
 
-        model.addAttribute("companyNumber", companyNumber);
-        model.addAttribute("penaltyNumber", penaltyId);
+        model.addAttribute(COMPANY_NUMBER_ATTR, companyNumber);
+        model.addAttribute(PENALTY_NUMBER_ATTR, penaltyId);
+        model.addAttribute(COMPANY_NAME_ATTR, companyProfileApi.getCompanyName());
+        model.addAttribute(REASON_ATTR, LATE_FILING_PENALTY_REASON);
+        model.addAttribute(PAYMENT_DATE_ATTR, setUpPaymentDateDisplay(payableLateFilingPenalty));
```

Comment:
Maybe change to 'payablePenalty'. I know the class is called 'PayableLateFilingPenalty' but might be better is we start to make these names more generic.

Reply:
Update as suggested